gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dcombobox_ttt2.lua

```diff
@@ -97,33 +126,94 @@ function PANEL:PerformLayout()
 end
 
 ---
--- @param string value
--- @param number index
+-- @param string title
+-- @param string/number value should be indexable
+-- @param bool select
+-- @param string icon
+-- @param any data additional data that you might want to use
+-- @return number index
 -- @realm client
-function PANEL:ChooseOption(value, index)
-	if self.menu then
-		self.menu:Remove()
-		self.menu = nil
+function PANEL:AddChoice(title, value, select, icon, data)
+	if istable(value) then
+		ErrorNoHalt("[TTT2] dcombobox_ttt2 AddChoice format changed to PANEL:AddChoice(title, value, select, icon, data)\n For any table data use the last parameter.\n")
+
+		return
 	end
 
-	self:SetText(value)
+	local index = #self.choices + 1
+
+	self.choices[index] = {
+		title = title,
+		value = value,
+		icon = icon,
+		data = data
+	}
+
+	-- Index tables for fast access
+	self.titleIndices[title] = index
+	self.valueIndices[value] = index
+
+	if select then
+		self:ChooseOption(nil, index, true)
+	end
+
+	return index
+end
+
+---
+-- Choose either value or index, title is set 
+-- @param[opt] string/number value should be indexable
+-- @param[opt] number index the option id
+-- @param[default=false] bool ignoreConVar To avoid endless loops, separated setting of convars and UI values
+-- @realm client
+function PANEL:ChooseOption(value, index, ignoreConVar)
+	if not value and not isnumber(index) then return end
+
+	if not index then
+		index = self:GetOptionId(value)
+
+		if not isnumber(index) then return end
+	end
 
 	self.selected = index
-	self:OnSelect(index, value, self.data[index])
+
+	self:SetText(self:GetOptionText(index))
+	self:OnSelect(index, self:GetOptionValue(index), self:GetOptionData(index))
```

Comment:
why is this using 3 functions to extract the data from the choice instead of just getting the choice and access the data directly?

Reply:
Also old structure
https://wiki.facepunch.com/gmod/DComboBox:OnSelect

Reply:
ah 3 functions because like this i can change the structure anytime. those are the getters